cleverpush/src/main/java/com/cleverpush/inbox/InboxDetailActivity.java

```diff
@@ -0,0 +1,574 @@
+  private ArrayList<Notification> notifications = new ArrayList<>();
+  private int selectedPosition = 0;
+  private String appBannerId = null;
+  private String appBannerNotificationId = null;
+  private Collection<AppBannersListener> bannersListeners = new ArrayList<>();
+  private Handler handler;
+  private String channel;
+  private boolean loading = false;
+  private Context context;
+  private Collection<Banner> banners = null;
+  private LinearLayout body;
+  private boolean isInitialized = false;
+  private Activity activity;
+  private ViewPager2 viewPager2;
+  private AppBannerOpenedListener openedListener;
+  private Banner data;
+  private View popupRoot;
+  private Handler mainHandler;
+
+  public static void launch(Activity activity, ArrayList<Notification> notifications, int selectedPosition) {
+    Intent intent = new Intent(activity, InboxDetailActivity.class);
+    intent.putExtra("notifications", notifications);
+    intent.putExtra("selectedPosition", selectedPosition);
+    activity.startActivity(intent);
+  }
+
+  public ActivityLifecycleListener getActivityLifecycleListener() {
+    return ActivityLifecycleListener.getInstance();
+  }
+
+  public Handler getHandler() {
+    return handler;
+  }
+
+  public boolean isLoading() {
+    return loading;
+  }
+
+  public void setLoading(boolean loading) {
+    this.loading = loading;
+  }
+
+  public Collection<Banner> getListOfBanners() {
+    return banners;
+  }
+
+  public AppBannerOpenedListener getOpenedListener() {
+    return openedListener;
+  }
+
+  public void setOpenedListener(AppBannerOpenedListener openedListener) {
+    this.openedListener = openedListener;
+  }
+
+  public Collection<AppBannersListener> getBannersListeners() {
+    return bannersListeners;
+  }
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_inbox_detail);
+
+    getSupportActionBar().setTitle("Message Detail");
+    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+    this.context = this;
+    this.activity = this;
+    init();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+  }
+
+  private void init() {
+    int layoutId = R.layout.activity_inbox_detail;
+    popupRoot = createLayout(layoutId);
+    handleBundleData(getIntent().getExtras());
+    handlerThread.start();
+    handler = new Handler(handlerThread.getLooper());
+    mainHandler = new CustomExceptionHandler(activity.getMainLooper());
+  }
+
+  private View createLayout(int layoutId) {
+    try {
+      View layout = activity.getLayoutInflater().inflate(layoutId, null);
+      layout.setOnClickListener(view -> dismiss());
+      return layout;
+    } catch (Exception exception) {
+      Logger.e(TAG, exception.getLocalizedMessage());
+    }
+    return null;
+  }
+
+  private void handleBundleData(Bundle extras) {
+    if (extras != null) {
+      if (extras.containsKey("selectedPosition")) {
```

Comment:
magic string, maybe create an enum for this and extras.containsKey("notifications")